httputil/cert_ext_tripper.go

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package httputil
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/asn1"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+
+	"github.com/hashicorp/go-hclog"
+)
+
+type ignoreExtensionsRoundTripper struct {
+	base         *http.Transport
+	extsToIgnore []asn1.ObjectIdentifier
+	logger       hclog.Logger
+}
+
+// NewIgnoreUnhandledExtensionsRoundTripper creates a RoundTripper that may be used in an HTTP client which will
+// ignore the provided extensions if present unhandled on a certificate.  If base is nil, the default RoundTripper is used.
+func NewIgnoreUnhandledExtensionsRoundTripper(logger hclog.Logger, base http.RoundTripper, extsToIgnore []asn1.ObjectIdentifier) http.RoundTripper {
```

Comment:
I don't love requiring a logger here as not all users of this library are necessarily using hclog (Boundary, for instance). Looking further down, it's used in one place purely for debugging, and in another to warn with the extension info when the verification is about to error out anyways. If the verification already tells you which extension is unhandled when it errors out (I'm not sure, as tls errors are fairly opaque in Go), I think we should just remove it entirely.

Reply:
Well, it's nilable and isn't used if not provided, but I'm not opposed to leaving it out.